src/Type.cpp

```diff
@@ -54,6 +54,7 @@
 #include "Probabilities.h"
 #include "DepthSpec.h"
 #include "Enumerator.h"
+#include "Function.h"
```

Comment:
Does not makes sense to include a `Function.h` here.

Reply:
It is required to include infrastructure defined in `Function.h` and even if I moved to a separate file I need to include whereever I want to use that infrastructure.

Reply:
Yep, please read my Review message, it's suggesting such a refactoring.